cartridges/int_turnto_core_v5/cartridge/scripts/jobsteps/TurnToFeedUpload.js

```diff
@@ -62,85 +72,109 @@ var run = function run() {
-                        // Localized siteKey and authKey, need to download per locale, added to the content body of the service request
-                        var ttSiteSet = siteAndAuthKeys.entrySet();
-                        var siteKey = null;
-                        var authKey = null;
-                        var domain = null;
-                        for (var ls = 0; ls < ttSiteSet.size(); ls++) {
-                            if (ttSiteSet[ls].value.locales.indexOf(locale) > -1) {
-                                siteKey = JSON.parse(ttSiteSet[ls].key);
-                                authKey = ttSiteSet[ls].value.authKey;
-                                domain = 'www.' + ttSiteSet[ls].value.domain;
-                                break;
-                            }
-                        }
-
-                        // If turntoAuthKey and turntoSiteKey values are not defined for a particular locale the job should skip the locale.
-                        if (empty(siteKey) || empty(authKey) || empty(domain)) {
-                            Logger.error('FAILED Site and/or Auth key or domain is missing: Site Key -> ' + siteKey + ' Auth Key -> ' + authKey + ' Export File Name -> ' + file.name + ' domain -> ' + domain);
+            try {
+                // List all files in the specific locale folder
+                var exportFiles = new File(turntoDir.getFullPath() + File.SEPARATOR + locale);
+                exportFiles = exportFiles.listFiles();
+
+                // If no files are found, go to the next locale
+                if (exportFiles) {
+                    // filter out files which match the file pattern
+                    var filteredExportFiles = exportFiles.clone();
+                    for (var k = 0; k < exportFiles.size(); k++) {
+                        var file = exportFiles[k];
+                        if (file.getName().match(filePattern) == null) {
+                            filteredExportFiles.remove(file);
                         } else {
-                            // feedUploadService
-                            var requestDataContainer = ServiceFactory.buildFeedUploadRequestContainer(postFileLocation, file, siteKey, authKey, domain, feedStyle);
-                            if (logging) {
-                                Logger.info('Post file location: ' + postFileLocation);
-                                Logger.info('Export file: ' + file);
-                            }
-
-                            // call service, returns success or error
-                            var feedUploadResult = FeedUploadService.call(requestDataContainer);
-
-                            if (feedUploadResult.error) {
-                                Logger.error('Feed upload service error: ' + feedUploadResult.error + ' ' + feedUploadResult.errorMessage);
+                            // Localized siteKey and authKey, need to download per locale, added to the content body of the service request
```

Comment:
To help with some of the nested ifs, instead of `else` here, you could put `continue;` in the `if` above.  Same for the `else` down on line 120.

Reply:
I started working on an update to replace the for loops with forEach(). I'll use "return" instead of "else" to clean this up. I need to get the conversion update out for a client so I'll add those updates to the next release.